store/userData.js

```diff
@@ -0,0 +1,21 @@
+import api from '@/services/githubService.js'
+
+export const state = () => ({
+  userData: {},
+})
+
+export const mutations = {
+  SET_USERDATA(state, userData){
+    state.userData = userData
+  },
+}
+
+export const actions = {
+  fetchUserData({commit}, username) {
+    return api.api(username).then(function(response) {
+      commit('SET_USERDATA', response.data)
+    })
```

Comment:
I am not a Nuxt expert but here I think we can have a "SET_USERDATA_ERROR" to catch the error as well.

```suggestion
    return api.api(username).then(function(response) {
      commit('SET_USERDATA', response.data)
    }).catch(function(error){
      commit('SET_USERDATA_ERROR', error);
    })
```

Reply:
Brilliant idea. I think this might also be the solution to my error handling difficulties.